src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,357 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { act, render } from '@testing-library/react';
+
+import * as AllContext from '../../../lib/components/app-layout/visual-refresh/context.js';
+import VisualRefreshTriggerButton, {
+  TriggerButtonProps as VisualRefreshTriggerButtonProps,
+} from '../../../lib/components/app-layout/visual-refresh/trigger-button';
+import VisualRefreshToolbarTriggerButton, {
+  TriggerButtonProps as VisualRefreshToolbarTriggerButtonProps,
+} from '../../../lib/components/app-layout/visual-refresh-toolbar/toolbar/trigger-button/index.js';
+import { ButtonProps } from '../../../lib/components/button';
+import { IconProps } from '../../../lib/components/icon/interfaces.js';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+import visualRefreshStyles from '../../../lib/components/app-layout/visual-refresh/styles.css.js';
+import toolbarTriggerButtonStyles from '../../../lib/components/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.css.js';
+
+const MockUseAppLayoutInternals = jest.spyOn(AllContext, 'useAppLayoutInternals').mockName('useAppLayoutInternals');
+
+const mockDrawerId = 'mock-drawer-id';
+const mockTestId = `awsui-app-layout-trigger-${mockDrawerId}`;
+
+const mockProps = {
+  ariaLabel: 'Aria label',
+  className: 'class-from-props',
+  iconName: 'bug',
+  testId: mockTestId,
+};
+const mockOtherEl = {
+  class: 'other-el-class',
+  text: 'other-element',
+};
+const mockUseLayoutInternalValues = {
+  hasOpenDrawer: false,
+  isMobile: false,
+};
+
+const mockEventBubble = {
+  bubbles: true,
+  isTrusted: true,
+  relatedTarget: null,
+};
+
+const mockEventBubbleWithShiftFocus = {
+  ...mockEventBubble,
+  relatedTarget: {
+    dataset: {
+      shiftFocus: 'last-opened-toolbar-trigger-button',
+    },
+  },
+};
+
+const delay = () => act(() => new Promise(resolve => setTimeout(resolve)));
+
+const renderVisualRefreshTriggerButton = async (
+  props: Partial<VisualRefreshTriggerButtonProps> = {},
+  useAppLayoutInternalsValues: Partial<AllContext.AppLayoutInternals> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  MockUseAppLayoutInternals.mockReturnValue({
+    ...mockUseLayoutInternalValues,
+    ...useAppLayoutInternalsValues,
+  } as AllContext.AppLayoutInternals);
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshTriggerButton {...(renderProps as VisualRefreshTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(visualRefreshStyles['trigger-wrapper'])!;
+  await delay();
```

Comment:
why is this delay needed?

Reply:
I was getting network errors, 404 of I did not have delays. I will see if it still applies

Reply:
I did not need. If i need later for the tooltip I will add a comment as to why. Thanks for pointing out